File: js/bible_pronunciation_choices_sound.mjs
import { arguments_assert } from "./arguments_assert.mjs";
import { folder_web_dev } from "./folder_web_dev.mjs";
import { path_join } from "./path_join.mjs";
import { bible_pronunciation_choices } from "./bible_pronunciation_choices.mjs";
import { fn_name } from "./fn_name.mjs";
import { py_script_speech_json_report } from "./py_script_speech_json_report.mjs";
export async function bible_pronunciation_choices_sound() {
  "Speaks both sayings of each name still in question into the dev sound folder, so the pair can be listened to one after the other on a phone.";
  "★ IT IS WRITTEN WHERE THE DEV SERVER ALREADY SERVES, WHICH IS THE WHOLE OF WHY THE PAIR CAN BE HEARD AT ALL. The person deciding is at a phone, not at this machine, so a file written anywhere else is a file nobody can play. The folder beside it holds the earlier listening tests for the same reason.";
  "★ ONE PROCESS SPEAKS BOTH SAYINGS, BECAUSE THE VOICE MODEL IS A THIRD OF A GIGABYTE. Reading it once and writing the name into the reader's dictionary between the two sayings costs one load; asking twice would cost fourteen.";
  arguments_assert(arguments, 0);
  let dev = folder_web_dev();
  let folder = path_join([dev, "sound_test"]);
  let choices = bible_pronunciation_choices();
  let script_name = fn_name("bible_pronunciation_choices_sound");
  let args = {
    folder,
    threads: 4,
    choices,
  };
  let reported = await py_script_speech_json_report(script_name, args);
  return reported;
}

File: scripts/py/bible_pronunciation_choices_sound.py
"""Speaks one Bible sentence twice for each name whose sounding is in question.

A name the dictionary now says differently from the way the recordings said it
is either a mend or a spoiling, and which one it is cannot be reasoned out: two
sayings of a name can both be defensible on paper and only one of them sound
right.  So the pair is put to the ear instead.  Each name gets the same sentence
spoken twice, once with the sounds the recording used and once with the sounds
the reading holds now, and the two files sit side by side under the same name.

★ THE SENTENCE IS A REAL VERSE AND THE WORD IS NOT LEFT STANDING ALONE.  A bare
word gets the prosody of a whole utterance - a falling end, a released final
consonant - and that is not the shape it is heard in on a recording, where it
sits in the middle of a line.  Two sayings compared as bare words can be told
apart by the wrong thing.  The verse is the same on both sides, so the sentence
around the name cancels and only the name is left to differ.

The sounds are written straight into the reader's dictionary between the two
sayings, which is what lets one process speak the same line both ways.  Nothing
is put back afterwards because nothing after this asks: the process speaks its
list and ends.
"""

import json
import sys

from text_to_speech import (
    COMPRESSION_LEVEL,
    SPEED,
    engine_ready,
    line_samples,
    said_text,
    write_mp3,
)


def choice_spoken(engine, folder, choice):
    """Writes the two mp3s for one name, and says where they went.

    The word is filed under the spelling the verse writes it with, which is the
    only spelling the reader looks the word up under.
    """
    g2p = engine["g2p"]
    word = choice["word"]
    sentence = said_text(choice["sentence"])
    written = {}
    for side in ("was", "now"):
        g2p.lexicon.golds[word] = choice[side]
        samples, rate = line_samples(g2p, engine["kokoro"], sentence, SPEED)
        path = folder + "/" + word + "_" + side + ".mp3"
        write_mp3(path, samples, rate, COMPRESSION_LEVEL)
        written[side] = path
    return written


def main(args_path):
    with open(args_path, encoding="utf-8") as fh:
        args = json.load(fh)
    folder = args["folder"]
    engine = engine_ready(args["threads"])
    written = {}
    for choice in args["choices"]:
        written[choice["word"]] = choice_spoken(engine, folder, choice)
    print(json.dumps({"folder": folder, "written": written}, ensure_ascii=False))


if __name__ == "__main__":
    main(sys.argv[1])
